File: sdk/src/main/java/org/maibot/sdk/util/UnwrapUtils.java
package org.maibot.sdk.util;

public class UnwrapUtils {
    public static <T> T unwrap(Class<T> clazz, Object... objects) {
        for (Object obj : objects) {
            if (clazz.isInstance(obj)) {
                return clazz.cast(obj);
            }
        }
        throw new ClassCastException("Cannot unwarp to " + clazz.getName());
    }
}

File: sdk/src/main/java/org/maibot/sdk/util/UnwrapUtils.kt
package org.maibot.sdk.util

object UnwrapUtils {
    @JvmStatic
    fun <T> unwrap(clazz: Class<T>, vararg objects: Any?): T {
        for (obj in objects) {
            if (clazz.isInstance(obj)) {
                return clazz.cast(obj)
            }
        }
        throw ClassCastException("Cannot unwarp to " + clazz.getName())
    }
}
